src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -900,29 +909,42 @@ private void populateDecoderControl(Map<String, String> stats, List<AdvancedCont
 		HDR hdr = decoderInfo.getHdrDynamicRange();
 		OutputFrameRate outputFrameRate = decoderInfo.getOutputFrameRate();
 		StillImage stillImage = decoderInfo.getStillImage();
-		String streamID = decoderInfo.getStreamId().toString();
-		if (StringUtils.isNullOrEmpty(streamID)) {
-			streamID = DecoderConstant.DEFAULT_STREAM_ID;
-		}
-		// Get list values of controllable property (dropdown)
-		List<String> outputFrameRateList = DropdownList.Names(OutputFrameRate.class);
-		List<String> stillImageList = DropdownList.Names(StillImage.class);
-		List<String> hdrList = DropdownList.Names(HDR.class);
-		List<String> streamIDList = new ArrayList<>();
 
+		Integer streamID = decoderInfo.getStreamId();
+		String streamName = DecoderConstant.NONE;
 		if (this.localStreamInfoList != null) {
-			for (StreamInfo streamInfo :
-					localStreamInfoList) {
-				streamIDList.add(streamInfo.getId().toString());
+			for (StreamInfo streamInfo : localStreamInfoList) {
+				if (streamID.equals(streamInfo.getId())) {
+					streamName = streamInfo.getName();
+					if (StringUtils.isNullOrEmpty(streamName)) {
+						streamName = streamInfo.getDefaultStreamName();
+					}
+					break;
+				}
 			}
-		} else {
-			streamIDList.add(DecoderConstant.DEFAULT_STREAM_ID);
 		}
 
+			// Get list values of controllable property (dropdown)
+			List<String> outputFrameRateList = DropdownList.Names(OutputFrameRate.class);
```

Comment:
It should be DropdownList.names instead of DropdownList.Names

Reply:
I fixed it